ci/run-clang-format.py

```
@@ -178,46 +179,41 @@ def run_clang_format_diff(args: Any, file: str) -> tuple[list[str], list[str]]:
+            invocation,
+            capture_output=True,
+            text=True,
+            encoding="utf-8",
+            errors="replace",
+        )
     except OSError as exc:
         raise DiffError(
             "Command '{}' failed to start: {}".format(
                 subprocess.list2cmdline(invocation), exc
             )
         )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)